apm-agent-plugins/apm-rabbitmq/apm-rabbitmq-spring/src/main/java/co/elastic/apm/agent/rabbitmq/MessageBatchIteratorWrapper.java

```diff
@@ -0,0 +1,69 @@
+package co.elastic.apm.agent.rabbitmq;
+
+
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import co.elastic.apm.agent.rabbitmq.header.SpringRabbitMQTextHeaderGetter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageProperties;
+
+import java.util.Iterator;
+
+public class MessageBatchIteratorWrapper implements Iterator<Message> {
+
+    public static final Logger logger = LoggerFactory.getLogger(MessageBatchIteratorWrapper.class);
+
+    private final Iterator<Message> delegate;
+    private final ElasticApmTracer tracer;
+    private final SpringAmqpTransactionHelper transactionHelper;
+
+    public MessageBatchIteratorWrapper(Iterator<Message> delegate, ElasticApmTracer tracer, SpringAmqpTransactionHelper transactionHelper) {
+        this.delegate = delegate;
+        this.tracer = tracer;
+        this.transactionHelper = transactionHelper;
+    }
+
+    @Override
+    public boolean hasNext() {
+        endCurrentTransaction();
+        return delegate.hasNext();
+    }
+
+    public void endCurrentTransaction() {
+        try {
+            Transaction transaction = tracer.currentTransaction();
+            if (transaction != null && "messaging".equals(transaction.getType())) {
+                transaction.deactivate().end();
+            }
+        } catch (Exception e) {
+            logger.error("Error in Spring AMQP iterator wrapper", e);
+        }
+    }
+
+    @Override
+    public Message next() {
+        endCurrentTransaction();
+
+        Message message = delegate.next();
+        try {
+            if (message == null) {
+                return message;
+            }
+            MessageProperties messageProperties = message.getMessageProperties();
+            if (messageProperties == null) {
+                return message;
+            }
```

Comment:
Why? We should create a transaction anyway - a root transaction if there is no `tracecontext` property, or child transaction if there is. We already do that (create a transaction when there is no property). The same goes for `SpringAmqpMessageListenerInstrumentation` - we should allow `null` properties.
Please fix `SpringAmqpTransactionHelper#createTransaction()` instead to allow for `null` properties (and treat the same as if there are properties without context property).

Reply:
fixed